lib/jekyll/commands/new.rb

```diff
@@ -28,19 +28,7 @@ def process(args, options = {})
                       "#{new_blog_path} exists and is not empty."
           end
 
-          if options["blank"]
-            create_blank_site new_blog_path
-          else
-            create_sample_files new_blog_path
-
-            File.open(File.expand_path(initialized_post_name, new_blog_path), "w") do |f|
-              f.write(scaffold_post_content)
-            end
-
-            File.open(File.expand_path("Gemfile", new_blog_path), "w") do |f|
-              f.write(gemfile_contents)
-            end
-          end
+          if_options_blank(new_blog_path, options)
```

Comment:
Why don't we have the `if` still in this method and move just the larger part of the `else` into its own method:

```ruby
def process(args, options = {})
  # ...
  if options["blank"]
    create_blank_site new_blog_path
  else
    create_site new_blog_path
  end

  # ...
end

def create_site(new_blog_path)
  create_sample_files new_blog_path

  File.open(...) do |f|
    # ...
  end

  File.open(...) do |f|
    # ..
  end
end
```

Does that sound good to you? Then the method doesn't start with an `if`.